test/prism_regression/hash.parse-tree.exp

```diff
@@ -0,0 +1,87 @@
+Begin {
+  stmts = [
+    Hash {
+      kwargs = false
+      pairs = [
+        Pair {
+          key = Symbol {
+            val = <U k1>
+          }
+          value = String {
+            val = <U v1>
+          }
+        }
+      ]
+    }
+    Hash {
+      kwargs = false
+      pairs = [
+        Pair {
+          key = Symbol {
+            val = <U k2>
+          }
+          value = String {
+            val = <U v2>
+          }
+        }
+      ]
+    }
+    Hash {
+      kwargs = false
+      pairs = [
+        Pair {
+          key = String {
+            val = <U k3>
+          }
+          value = String {
+            val = <U v3>
+          }
+        }
+      ]
+    }
+    DefMethod {
+      name = <U k4>
+      args = NULL
+      body = String {
+        val = <U Method for implicit value>
+      }
+    }
+    Hash {
+      kwargs = false
+      pairs = [
+        Pair {
+          key = Symbol {
+            val = <U k4>
+          }
+          value = Send {
+            receiver = NULL
+            method = <U k4>
+            args = [
+            ]
+          }
+        }
+      ]
+    }
```

Comment:
This is the output the code currently produces, which seems correct, but doesn't match what the legacy parser produces. This is the diff that it suggests:

```diff
@@ -53,7 +53,11 @@
           key = Symbol {
             val = <U k4>
           }
-          value = Self {
+          value = Send {
+            receiver = NULL
+            method = <U k4>
+            args = [
+            ]
           }
         }
       ]
```

I have no idea why it's suggesting that. Where did `self` come from?

One complication here is that parsing Ruby correctly also requires you keep context of what local variables have been declared so far (so you can emit lvar lookups or method calls, as needed). Perhaps the legacy parser didn't do this during parsing, and resolved it later? IDK, that's a monday problem.

Reply:
Pulling this out of the PR for now, will address by #104